InterfaceTPL/tplUnitTests.cpp

```diff
@@ -0,0 +1,139 @@
+ * IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * UT-BATTELLE, LLC AND THE UNITED STATES GOVERNMENT MAKE NO REPRESENTATIONS AND DISCLAIM ALL WARRANTIES, BOTH EXPRESSED AND IMPLIED.
+ * THERE ARE NO EXPRESS OR IMPLIED WARRANTIES OF MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE, OR THAT THE USE OF THE SOFTWARE WILL NOT INFRINGE ANY PATENT,
+ * COPYRIGHT, TRADEMARK, OR OTHER PROPRIETARY RIGHTS, OR THAT THE SOFTWARE WILL ACCOMPLISH THE INTENDED RESULTS OR THAT THE SOFTWARE OR ITS USE WILL NOT RESULT IN INJURY OR DAMAGE.
+ * THE USER ASSUMES RESPONSIBILITY FOR ALL LIABILITIES, PENALTIES, FINES, CLAIMS, CAUSES OF ACTION, AND COSTS AND EXPENSES, CAUSED BY, RESULTING FROM OR ARISING OUT OF,
+ * IN WHOLE OR IN PART THE USE, STORAGE OR DISPOSAL OF THE SOFTWARE.
+ */
+
+#ifndef __TPL_UNIT_TESTS_CPP
+#define __TPL_UNIT_TESTS_CPP
+
+#include "tplUnitTests.hpp"
+#include "tsgBlasWrappers.hpp"
+#include <string>
+#include <math.h>
+#include <cmath>
+
+bool testTridiagonal();
+
+TplUnitTester::TplUnitTester() : verbose(false){}
+TplUnitTester::~TplUnitTester() {}
+
+void TplUnitTester::setVerbose(bool new_verbose){ verbose = new_verbose; }
+
+bool TplUnitTester::Test(UnitTests test){
+    if (test == unit_all) {
+        cout << endl << endl;
+        cout << "---------------------------------------------------------------------" << endl;
+        cout << "       Tasmanian Third Party Libraries (TPL): Unit Tests" << endl;
+        cout << "---------------------------------------------------------------------" << endl << endl;
+        bool pass = true;
+        pass = testTriToeplitzEigen();
+        cout << endl;
+        if (pass){
+            cout << "---------------------------------------------------------------------" << endl;
+            cout << "           All Unit Tests Completed Successfully" << endl;
+            cout << "---------------------------------------------------------------------" << endl << endl;
+        } else {
+            cout << "FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL" << endl;
+            cout << "         Some Unit Tests Have Failed" << endl;
+            cout << "FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL" << endl << endl;
+        }
+        return pass;
+    } else if (test == unit_none) {
+        cout << "No tests to run.\n";
+        return true;
+    } else {
+        cout << "Unknown test type!\n";
+        return false;
+    }
+}
+
+bool testTriToeplitzEigen() {
```

Comment:
Move this to `tasgridUnitTests.cpp` and add a new tests. You can use the existing `doesMatch()` methods.

Reply:
Done.